tests/common.py

```diff
@@ -207,19 +213,69 @@ def assert_vm_image(self, exp: VMImage, act: VMImage):
         self.assertEqual(exp.firmwareType, act.firmwareType, "VM_Image.firmwareType")
         self.assertEqual(exp.hwRngTypeOfImage, act.hwRngTypeOfImage, "VM_Image.hwRngTypeOfImage")
 
+    def check_installation_requirement(
+        self, ob_1: InstantiationRequirement, ob_2: InstantiationRequirement
+    ):
+        self.assert_gaia_x_entity(ob_1, ob_2)
+
+    def check_hypervisor(self, ob_1: Hypervisor, ob_2: Hypervisor):
+        self.assert_software_resource(ob_1, ob_2)
+        self.assertEqual(
+            ob_1.hypervisorType, str(ob_2.hypervisorType), "Hypervisor.hypervisorType")
+
+    def assert_flavor(self, ob_1: ServerFlavor, ob_2: ServerFlavor):
+        # self.check_installation_requirement(ob_1, ob_2)
+        self.assert_cpu(ob_1.cpu, ob_2.cpu)
+        self.assert_mem(ob_1.ram, ob_2.ram)
+        self.assert_gpu(ob_1.gpu, ob_2.gpu)
+        self.assert_disk(ob_1.bootVolume, ob_2.bootVolume)
+        self.assertEqual(
+            len(ob_1.additionalVolume),
+            len(ob_2.additionalVolume),
+            "ServerFlavor.additionalVolume",
+        )
+        if ob_1.hypervisor:
+            self.check_hypervisor(ob_1.hypervisor, ob_2.hypervisor)
+        if ob_1.confidentialComputing:
+            self.assertEqual(
+                ob_1.confidentialComputing,
+                str(ob_2.confidentialComputing),
+                "ServerFlavor.confidentialComputing",
+            )
+        self.assertEqual(
+            ob_1.hardwareAssistedVirtualization,
+            ob_2.hardwareAssistedVirtualization,
+            "ServerFlavor.hardwareAssistedVirtualization",
+        )
+        self.assertEqual(
+            ob_1.hwRngTypeOfFlavor,
+            ob_2.hwRngTypeOfFlavor,
+            "ServerFlavor.hwRngTypeOfFlavor",
+        )
+
+        for i in range(9, len(ob_1.additionalVolume) - 1):
+            self.assert_disk(ob_1.additionalVolume[i], ob_2.additionalVolume[i])
 
 class MockConnection:
     """
     Wrap connection to OpenStack Cluster
     """
 
-    images = []
-
-    def __init__(self, images: List[Image]):
+    def __init__(self, images: List[OS_Image] = None, flavors: List[OS_Flavor] = None):
         self.images = images
+        self.flavors = flavors
```

Comment:
```suggestion
        self.images = images or ()
        self.flavors = flavors or ()
```

Reply:
Why brackets? self.images and self.flavors are lists. Mustn't it be square brackets?

 ```suggestion
        self.images = images or []
        self.flavors = flavors or []
```

Reply:
NB, I think square brackets is a tautology; as far as I know, we have parentheses `()`, brackets `[]`, braces `{}` and angular brackets `<>` (but this may just be my own nomenclature)

Reply:
I prefer `[]` as immutability of images and flavors is no a requirement.

Reply:
Done in 0012be6390ae38077008b8c5ca702a67dccb877a